webapp/src/lib/components/spot-component/spot-location-create-form.svelte

```diff
@@ -5,100 +5,280 @@
-        labelText="Street address"
-        name="street-address"
-        placeholder="Street Address"
-        readonly={isReadOnly}
-        bind:value={streetAddress}
-    />
-    <TextInput
-        required
-        labelText="City"
-        name="city"
-        placeholder="City"
-        readonly={isReadOnly}
-        bind:value={city}
-    />
-    <Select
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        labelText="Province"
-        bind:selected={state}
-    >
-        {#each provinceData.keys() as key}
-            <SelectItem value={key} text={provinceData.get(key)} />
-        {/each}
-    </Select>
-
-    <Select
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        labelText="Country Code"
-        bind:selected={countryCode}
-    >
-        <SelectItem value="CA" text="Canada" />
-    </Select>
-    <TextInput
-        required
-        labelText="Postal code"
-        name="postal-code"
-        placeholder="Postal code"
-        readonly={isReadOnly}
-        bind:value={postalCode}
-    />
-    <p>Utilities</p>
-    <Checkbox
-        name="shelter"
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        labelText="shelter"
-        bind:checked={hasShelter}
-    />
-    <Checkbox
-        name="plug-in"
-        labelText="Plug-in"
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        bind:checked={hasPlugIn}
-    />
-    <Checkbox
-        name="charging-station"
-        labelText="Charging Station"
-        style={isReadOnly ? 'pointer-events: none;' : ' '}
-        bind:checked={hasChargingStation}
-    />
-    {#if !isReadOnly}
-        <Button type="submit">Submit</Button>
-    {/if}
-</Form>
+<!-- svelte-ignore a11y-click-events-have-key-events a11y-no-static-element-interactions-->
+<div on:click={handleClickOutside}>
+    <Form on:submit={handleSubmit}>
+        {#if !isReadOnly}
+            <Search
+                type="text"
+                bind:value={searchQuery}
+                placeholder="Enter an address... [At least 3 characters required]"
+                name="search"
+                on:input={handleInput}
+            />
+            {#if results?.length > 0 && dropdownOpen}
+                <ul class="dropdown">
+                    {#each results as result}
+                        <!-- svelte-ignore a11y-click-events-have-key-events a11y-no-noninteractive-element-interactions -->
+                        <li on:click={() => handleSelect(result)}>
+                            {result.properties.formatted}
+                        </li>
+                    {/each}
+                </ul>
+            {/if}
+        {/if}
+        <Modal
+            open={warningModalOpen}
+            modalHeading="Error - Please check Address"
+            primaryButtonText="Re-enter address"
+            on:close
+            on:open
+            on:submit={() => (warningModalOpen = false)}
+        >
+            {warningText}
+        </Modal>
```

Comment:
Maybe try moving it to +page.svelte and use this as the error display mechanism in place of the current toast message?

Reply:
I didn't want to return an error per se as in future if we do include support in other countries the error would be of no use. (And if you think about it, it actually isn't an error, just a unsupported request at the moment)